### PR TITLE
Refactor: reporting requirement database refactor part 1

### PR DESCRIPTION
**Pull request by dleard**
issue #761
Part 1 of a refactor. The changes in part 1 should not affect the functionality of the app. Part 2 will have cascading effects.

### CODE REVIEW

File: schema/deploy/tables/emission_intensity_report.sql [hunk 1]
```diff
@@ -0,0 +1,57 @@
+-- Deploy cif:tables/emission_intensity_report to pg
+-- requires: tables/reporting_requirement
+
+-- Deploy cif:tables/emission_intensity_report to pg
+
+begin;
+
+create table cif.emission_intensity_report(
+  id integer primary key generated always as identity,
+  reporting_requirement_id integer references cif.reporting_requirement(id),
```
Comment (matthieu-foucault) [inline schema/deploy/tables/emission_intensity_report.sql:10]:
```suggestion
  reporting_requirement_id integer not null references cif.reporting_requirement(id),
```

File: schema/deploy/tables/emission_intensity_report.sql [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  measurement_period_start_date timestamptz not null,
+  measurement_period_end_date timestamptz not null,
+  emission_functional_unit varchar(100) not null default 'tCO2e',
+  production_functional_unit varchar(100) not null,
+  baseline_emission_intensity numeric not null,
+  target_emission_intensity numeric not null,
+  post_project_emission_intensity numeric,
+  total_lifetime_emission_reduction numeric,
+  adjusted_emissions_intensity_performance numeric
```
Comment (matthieu-foucault) [inline schema/deploy/tables/emission_intensity_report.sql:19]: I would be more permissive and keep all of these nullable

File: schema/deploy/tables/milestone_report.sql [hunk 3]
```diff
@@ -0,0 +1,49 @@
+-- Deploy cif:tables/milestone_report to pg
+-- requires: tables/reporting_requirement
+
+begin;
+
+create table cif.milestone_report(
+  id integer primary key generated always as identity,
+  reporting_requirement_id integer references cif.reporting_requirement(id),
+  substantial_completion_date timestamptz not null,
```
Comment (matthieu-foucault) [inline schema/deploy/tables/milestone_report.sql:9]:
```suggestion
  substantial_completion_date timestamptz,
```

File: schema/deploy/tables/milestone_report.sql [hunk 4]
```diff
@@ -0,0 +1,49 @@
+-- Deploy cif:tables/milestone_report to pg
+-- requires: tables/reporting_requirement
+
+begin;
+
+create table cif.milestone_report(
+  id integer primary key generated always as identity,
+  reporting_requirement_id integer references cif.reporting_requirement(id),
```
Comment (matthieu-foucault) [inline schema/deploy/tables/milestone_report.sql:8]:
```suggestion
  reporting_requirement_id integer not null references cif.reporting_requirement(id),
```

File: schema/deploy/tables/report_type.sql [hunk 5]
```diff
@@ -5,7 +5,9 @@ begin;
 
 create table cif.report_type
 (
-  name varchar(1000) primary key
+  name varchar(1000) primary key,
+  form_schema jsonb,
```
Comment (matthieu-foucault) [inline schema/deploy/tables/report_type.sql:9]: are we planning to use this? If we were to store form schemas in the db, I'd have them in a separate table instead

**Review by matthieu-foucault — CHANGES_REQUESTED** [review]

File: schema/deploy/tables/report_type.sql (hunk 5 above)
Comment (dleard) [reply schema/deploy/tables/report_type.sql:9]: yeah, this should be a foreign key that's added whenever we implement versioned form schemas. I'll remove it here and take it out of the data model for now.

File: schema/deploy/tables/payment.sql [hunk 6]
```diff
@@ -5,13 +5,11 @@ begin;
...
-  date_paid timestamptz,
-  comment varchar(10000),
-  transaction_id varchar(1000),
-  status varchar(1000),
-  reporting_requirement_id integer references cif.reporting_requirement(id) not null
+  reporting_requirement_id integer references cif.reporting_requirement(id) not null,
+  gross_amount numeric not null,
+  net_amount numeric not null,
+  date_sent_to_csnr timestamptz,
+  comment varchar(10000)
```
[the remark this answers is not in the document]
Comment (dleard) [reply schema/deploy/tables/payment.sql]: I was struggling with whether payment should have an fkey to reporting_requirement or the other way around? If reporting_requirement is the one with the fkey, it'll have to be nullable

[the remark this answers is not in the document]
Comment (dleard) [reply schema/deploy/tables/payment.sql]: I thought maybe payments could relate to something other than reporting_requirement in the future which is why I currently have the fkey on reporting_requirement

[the remark this answers is not in the document]
Comment (dleard) [reply schema/deploy/tables/payment.sql]: Nevermind, I think the payment should have the fkey, not reporting_requirement. I just had to type it out and read it apparently in order to figure it out

[the remark this answers is not in the document]
Comment (dleard) [reply schema/deploy/tables/payment.sql]: ^ I'm going to handle this in Part 2

**Review by matthieu-foucault — APPROVED** [review]

**Comment by matthieu-foucault** [comment]
@dleard this needs a rebase

**Review by matthieu-foucault — APPROVED** [review]

**Pull request merged**